### ISSUE TITLE
Split package warning when using reactive messaging with in memory connector for tests

### ISSUE THREAD

**Issue by mswiderski** [quarkusio/quarkus#22837]
### Describe the bug

When using reactive messaging (e.g. kafka or MQTT) and in memory connector for testing there is a warning logged on start
````
2022-01-12 14:39:26,821 WARN  [io.qua.arc.dep.SplitPackageProcessor] (build-4) Detected a split package usage which is considered a bad practice and should be avoided. Following packages were detected in multiple archives:
- "io.smallrye.reactive.messaging.providers.connectors" found in [io.smallrye.reactive:smallrye-reactive-messaging-in-memory::jar, io.smallrye.reactive:smallrye-reactive-messaging-provider::jar]
````

### Expected behavior

No warning should be logged 

### Actual behavior

Split package warning should not be displayed

### How to Reproduce?

include following dependency into a project with reactive messaging and the warning should show up

````
    <dependency>
      <groupId>io.smallrye.reactive</groupId>
      <artifactId>smallrye-reactive-messaging-in-memory</artifactId>
      <scope>test</scope>
    </dependency>
````

### Output of `uname -a` or `ver`

Darwin MacBook-Pro.local 21.1.0 Darwin Kernel Version 21.1.0: Wed Oct 13 17:33:23 PDT 2021; root:xnu-8019.41.5~1/RELEASE_X86_64 x86_64

### Output of `java -version`

openjdk version "11.0.11" 2021-04-20

### Quarkus version or git rev

2.6.2.Final

### Build tool (ie. output of `mvnw --version` or `gradlew --version`)

Apache Maven 3.6.3

**Comment by cescoffier** [comment]
This would need to be fixed in Smallrye.

**Comment by cescoffier** [comment]
@ozangunalp do you think we could fix it before Quarkus 3.0?

**Comment by cescoffier** [comment]
Work started upstream.
